specification/columns/commitmentdiscountconsumedquantity.md

```diff
@@ -0,0 +1,37 @@
+# Commitment Discount Consumed Quantity
+
+Commitment Discount Consumed Quantity is the numeric value of [CommitmentDiscountUnits](#commitmentdiscountunit) that is either applied to a metered [*resource*](#glossary:resource) or [*service*](#glossary:service) or is unused over a [*charge period*](#glossary:chargeperiod).
+
+CommitmentDiscountConsumedQuantity MUST be present in the billing data when the provider supports *commitment-based discounts*. When CommitmentDiscountStatus is "Used", CommitmentDiscountConsumedQuantity MUST be the amortized quantity of a corresponding *commitment-based discount* by a metered *resource* or *service* over the charge period. When CommitmentDiscountStatus is "Unused", CommitmentDiscountConsumedQuantity MUST be the remaining, unused quantity for the corresponding *commitment-based-discount* over the charge period. CommitmentDiscountConsumedQuantity values MUST be less than or equal to the corresponding ConsumedQuantity value over a charge period when both values are not null and ConsumedUnit and CommitmentDiscountUnit values are equal.
+
+CommitmentDiscountConsumedQuantity MUST be null when [ChargeCategory](#chargecategory) is *Purchase*, [PricingCategory](#pricingcategory) is *Committed*, and [*CommitmentDiscountId*](#commitmentdiscountid) is not null.
+
+CommitmentDiscountConsumedQuantity MUST be a positive value when [ChargeCategory](#chargecategory) is *Usage*, [PricingCategory](#pricingcategory) is *Committed*, [ResourceId](#resourceid) is not null, and [CommitmentDiscountId](#commitmentdiscountid) is not null.
+
+CommitmentDiscountConsumedQuantity MUST be null in all other cases.
```

Comment:
```suggestion
The CommitmentDiscountConsumedQuantity column adheres to the following requirements:

 * CommitmentDiscountConsumedQuantity MUST be present in the billing data when the provider supports *commitment-based discounts*.
 * CommitmentDiscountConsumedQuantity MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements.
 * CommitmentDiscountConsumedQuantity MUST NOT be null if [ChargeCategory](#chargecategory) is "Usage" and [*CommitmentDiscountId*](#commitmentdiscountid) is not null, unless [ChargeClass](#chargeclass) is "Correction".
 * CommitmentDiscountConsumedQuantity MAY be null if ChargeClass is "Correction".
 * CommitmentDiscountConsumedQuantity MUST be null in all other cases.
 * CommitmentDiscountConsumedQuantity value MAY be negative in cases where ChargeClass is "Correction".
 * When CommitmentDiscountStatus is "Used", CommitmentDiscountConsumedQuantity MUST match the amortized quantity of the corresponding *commitment-based discount* over the charge period.
 * When CommitmentDiscountStatus is "Unused", CommitmentDiscountConsumedQuantity MUST match the remaining, unused quantity of the corresponding *commitment-based discount* over the charge period.
```
I hope that by rephrasing this paragraph, I've managed to capture all the requirements, but I definitely need a second pair of eyes. :)

**Notes:** 

 - The following requirement is currently excluded, as the amortization factor can vary, potentially resulting in CommitmentDiscountConsumedQuantity being either greater than or less than ConsumedQuantity even if the Units match.

> CommitmentDiscountConsumedQuantity values MUST be less than or equal to the corresponding ConsumedQuantity value over a charge period when both values are not null and ConsumedUnit and CommitmentDiscountUnit values are equal.

- The following requirement is also currently excluded, as I'm not sure if it is correct and would like to discuss it

> CommitmentDiscountConsumedQuantity MUST be null when [ChargeCategory](#chargecategory) is *Purchase*, [PricingCategory](#pricingcategory) is *Committed*, and [*CommitmentDiscountId*](#commitmentdiscountid) is not null.

Reply:
"CommitmentDiscountConsumedQuantity MUST be null when [ChargeCategory](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/400#chargecategory) is Purchase, [PricingCategory](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/400#pricingcategory) is Committed, and [CommitmentDiscountId](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/400#commitmentdiscountid) is not null." --> Denotes that the `CommitmentDiscountConsumedQuantity` must be null for a purchase record.  Is this the problematic clause, "[CommitmentDiscountId](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/400#commitmentdiscountid) is not null."?

Reply:
@ijurica Can you provide an example for this?

<img width="785" alt="image" src="https://github.com/user-attachments/assets/2cac11fd-dd0f-4909-9d89-c06d7e84cb14">